nixos/modules/services/networking/networkmanager.nix

```diff
@@ -227,13 +215,30 @@ in {
         '';
       };
 
-      packages = mkOption {
+      basePackages = mkOption {
+        type = types.listOf types.package;
+        default = with pkgs; [
+          modemmanager
+          networkmanager
+          networkmanager-fortisslvpn
+          networkmanager-iodine
+          networkmanager-l2tp
+          networkmanager-openconnect
+          networkmanager-openvpn
+          networkmanager-vpnc
+          networkmanager-sstp
+        ];
+        description = ''
+          Basic packages of networkmanager, can be set to have a more minimal networkmanager.
+        '';
+      };
+
+     extraPackages = mkOption {
```

Comment:
Wait, now I got to this point and I'm a bit confused, why are all those plugins in the base instead of here?

Reply:
just for backwards compatibility. so people don't get missing packages if they don't change anything. the old packages option was really an extraPackages option and if we set a default here, it will override the default list.